packages/codemod/src/utils/prop-utils.ts

```diff
@@ -1,31 +1,129 @@
-import { ASTPath, JSCodeshift, JSXIdentifier, JSXOpeningElement } from "jscodeshift";
+import {
+  ASTPath,
+  Collection,
+  JSCodeshift,
+  JSXAttribute,
+  JSXElement,
+  JSXIdentifier,
+  JSXOpeningElement,
+  Literal
+} from "jscodeshift";
+import { logPropMigrationError } from "./report-utils";
 
 /**
  * Updates a prop name in a JSX element
  */
 export function updatePropName(
-  elementPath: ASTPath<JSXOpeningElement>,
+  j: JSCodeshift,
+  elementPath: ASTPath<JSXElement>,
   propsNamesMappingOldToNew: Record<string, string>
 ): void {
-  const attributes = elementPath.node?.attributes;
-  if (!attributes) return;
-
-  attributes.forEach(attr => {
-    if (
-      attr.type === "JSXAttribute" &&
-      attr.name.type === "JSXIdentifier" &&
-      propsNamesMappingOldToNew[attr.name.name]
-    ) {
-      attr.name.name = propsNamesMappingOldToNew[attr.name.name];
-    }
-  });
+  j(elementPath)
+    .find(JSXOpeningElement)
+    .find(JSXIdentifier)
+    .forEach(identifierPath => {
+      if (propsNamesMappingOldToNew[identifierPath.node.name]) {
```

Comment:
since we're inside "JSXIdentifier", we'll always have node.name, otherwise we won't get into the "forEach"
the previous way was not a jscodeshift-way, but instead manipulating the node, this way is more readable and efficient